cyences_app_for_splunk/bin/sophos_instance_details.py

```diff
@@ -0,0 +1,206 @@
+                    response_json = response.json()
+                    if response.status_code == 200:
+                        for instance in response_json.get("items"):
+                            list_of_uuid.append(instance)
+                    else:
+                        raise Exception("Error while fetching UUID of instance")
+
+            return list_of_uuid
+        except Exception as e:
+            self.logger.exception("Error while fetching Instance UUID : {}".format(e))
+
+
+
+    def generate(self):
+
+        try:
+            ip = self.ip
+            hostname = self.hostname
+            uuid = self.uuid
+            all_endpoints = self.all_endpoints
+            if(ip=="" and hostname=="" and uuid=="" and ip==" " and hostname==" " and uuid==" " and all_endpoints==False):
+                raise Exception("Please provide IP or Hostname or UUID")
+
+
+            # Read Client ID & Client Secret
+            client_id,client_secret = self.get_client_details()
+
+            if not client_id or not client_secret:
+                raise Exception("Sophos Client ID or Client Secret or both not found in the cs_configurations.conf file.")
+            else:
+                barier_token = self.get_barier_token(client_id,client_secret)
+                self.get_tenant_list(barier_token)
+
+                if(all_endpoints):
+                    for i in SOPHOS_TENANT_DICT:
+
+                        current_page=1
+                        total_page = 1
+                        while(current_page<=total_page):
+                            requestHeaders = self.get_request_header(i,barier_token)
+                            if(current_page==1):
+                                requestUrl = SOPHOS_TENANT_DICT.get(i)+"/endpoint/v1/endpoints?pageTotal=true"
+                            else:
+                                requestUrl =  SOPHOS_TENANT_DICT.get(i)+"/endpoint/v1/endpoints?page="+str(current_page)
+                            request = requests.get(requestUrl, headers=requestHeaders)
+                            if(request.status_code!=200):
+                                raise Exception("Error while fetching all the endpoints")
+                            request_json = request.json()
+
+                            if(current_page==1):
+                                total_page = request_json.get("pages").get("total")
+
+                            for i in request_json.get("items"):
+                                yield i
+                            
+                            current_page = current_page + 1
+
+                else:
+                    instance_details = self.get_instance_uuid(barier_token,ip,hostname,uuid)
+                    for i in instance_details:
+                        yield i
+        except Exception as e:
+            self.logger.exception("Error Occured while fetching instance details. Error: {}".format(e))
```

Comment:
import logging
import logger_manager
logger = logger_manager.setup_logging('sophos_details_command', logging.DEBUG)

Use **logger** instead of **self.logger** through-out code.

Reply:
Also, add 
**raise e**
so, we'll get notified in the UI.